server/pulp/server/controllers/repository.py

```diff
@@ -0,0 +1,111 @@
+import logging
+
+from pulp.common import dateutils
+from pulp.plugins.loader import api as plugin_api
+from pulp.server.db import connection, model
+
+_logger = logging.getLogger(__name__)
+
+
+def find_repo_content_units(
+        repository, repo_content_unit_q=None,
+        units_q=None, units_fields=None, limit=None, skip=None):
+    """
+    Search content units associated with a given repository
+
+    Each association returned will have a parameter "unit" set to the actual unit
+
+    :param repository: The repository to search.
+    :type repository: pulp.server.db.model.Repository
+    :param repo_content_unit_q: Any query filters to apply to the RepoContentUnits.
+    :type repo_content_unit_q: mongoengine.Q
+    :param units_q: Any query filters to apply to the ContentUnits.
+    :type units_q: mongoengine.Q
+    :param units_fields: List of fields to fetch for the unit objects, defaults to all fields
+    :type units_fields: List of str
+    :param limit: The maximum number of items to return for the given query.
+    :type limit: int
+    :param skip: The starting offset.
+    :type skip: int
+    :return: Content unit assoociations matching the query.
```

Comment:
Consider adding a newline above this to separate the `return` and `rtype` from the input params.